### DISCUSSION TITLE
Could syntax like this allow for nested partial types to be possible without the parent type being partial?

### DISCUSSION THREAD

**Discussion by nathanAjacobs** [dotnet/csharplang#7595] [Language Ideas]
The following syntax would allow source generators to support nested types easily without requiring the parent class to be declared partial.

```cs
// in File1.cs
namespace MyNamespace;

public class ParentClass
{
    private partial class NestedClass
    {
    }
}

// in File2.cs
namespace MyNamespace;

private partial class ParentClass.NestedClass
{

}
```

**Comment by CyrusNajmabadi** [comment]
Our view is that types must be marked partial. That way things cannot be added to them without that being an explicitly opted into thing.

**Reply by nathanAjacobs** [reply]
Would this not still satisfy that contract though? The only type that would be added to is the `NestedClass` which is marked partial.

The keywords would only apply to the type after the last '.'

**Reply by CyrusNajmabadi** [reply]
You'd be adding a member to the outer type in another file.  We intentionally make that something the outer type needs to opt into :-)

**Reply by jnm2** [reply]
@CyrusNajmabadi I'd say that the outer type already opts into this, at the granularity of the nested type only, by virtue of the fact of including the `partial` keyword on its declaration of the nested type.

**Reply by CyrusNajmabadi** [reply]
oh, i see.  i missed that the original decalration was opting in.  That does change things a bit.  That said, the motivation of "would allow source generators to support nested types easily" is less compelling to me as i don't think it's particularly hard to get SGs to support nested types in the first place.

**Reply by nathanAjacobs** [reply]
@CyrusNajmabadi It can be done with source generators currently, but it requires the whole type tree to be partial, whereas this would only require the nested type to be partial.
